Sources/SwiftFusion/Inference/BearingRangeFactor.swift

```diff
@@ -0,0 +1,193 @@
+// Copyright 2019 The SwiftFusion Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+import TensorFlow
+//
+///// Protocol for indicating a struct is range-capable
+//public protocol HasRange: Differentiable {
+//  associatedtype A: Differentiable
+//  associatedtype B: Differentiable where B.TangentVector: EuclideanVectorSpace & VectorConvertible & AdditiveArithmetic
```

Comment:
You should be able to remove all these constraints, except for `Differentiable` from `B`. This is because `Differentiable` is the only constraint that you need for a differentiable `range` function to make sense. This should make it easier to make things conform.

Of course, the `BearingFactor` will need more constraints on the type, but the `BearingFactor` can specify those constraints.